### PR TITLE
Some enhancements

### PR DESCRIPTION
**Pull request by andrey-kolegov**
+ Logging ability via standard golang logger interface.
+ Verbosef func for formatted output
* WaitForDigit modified to return an empty string when timeout

### CODE REVIEW

File: agi.go [hunk 1]
```diff
@@ -408,11 +434,37 @@ func (a *AGI) StreamFile(name string, escapeDigits string, offset int) (digit st
...
+func (a *AGI) Verbosef(format string, args ...interface{}) error {
+	return a.Verbose(fmt.Sprintf(format, args...), 9)
 }
 
 // WaitForDigit waits for a DTMF digit and returns what is received
 func (a *AGI) WaitForDigit(timeout time.Duration) (digit string, err error) {
 	resp := a.Command("WAIT FOR DIGIT", toMSec(timeout))
-	return string(resp.Result), resp.Error
+	resp.ResultString = ""
+	if resp.Error == nil && resp.Result >= 32 {
```
Comment (Ulexus) [inline agi.go]: I would prefer to see this using [`strconv.IsPrint`](https://golang.org/pkg/strconv/#IsPrint), for formalism.

File: agi.go [hunk 2]
```diff
@@ -408,11 +434,37 @@ func (a *AGI) StreamFile(name string, escapeDigits string, offset int) (digit st
 
 // Verbose logs the given message to the verbose message system
 func (a *AGI) Verbose(msg string, level int) error {
-	return a.Command("VERBOSE", msg, strconv.Itoa(level)).Err()
+	return a.Command("VERBOSE", strconv.Quote(msg), strconv.Itoa(level)).Err()
+}
+
+// Verbosef logs the formatted verbose output
+func (a *AGI) Verbosef(format string, args ...interface{}) error {
```
Comment (Ulexus) [inline agi.go:441]: +1

**Review by Ulexus — COMMENTED** [review]
Good ideas

**Comment by Ulexus** [comment]
You'll need to rebase, since I had already merged your previous Verbose fix

File: agi.go (hunk 1 above)
Comment (andrey-kolegov) [reply agi.go]: Agree.

**Comment by andrey-kolegov** [comment]
Resolved?

**Comment by Ulexus** [comment]
Can you fix make the IsPrint change?  Then we should be good.  Thanks

**Comment by andrey-kolegov** [comment]
Done.

**Comment by Ulexus** [comment]
Thanks!

**Pull request merged**